internal/backup/volumesnapshotcontent_action.go

```diff
@@ -153,10 +153,11 @@ func (p *VolumeSnapshotContentBackupItemAction) Progress(operationID string, bac
 	if boolptr.IsSetToTrue(vsc.Status.ReadyToUse) {
 		progress.Completed = true
 	} else if vsc.Status.Error != nil {
-		progress.Completed = true
+		errorMessage := ""
```

Comment:
If we don't mark the `progress.Completed` as `true` when there is an error in the VSC and VS status, one possible side-effect is that if the error cannot be resolved by retrying, the action will not be completed until the timeout. The default async operation timeout is 4 hours.

Reply:
Can we add a mechanism to count down the failure limitation?
For example, we can set a max try time for the VSC on error, and put that information in the VSC annotation, then update the error tried number accordingly. Until the limitation is hit, fail the action.

Reply:
I am not sure how to track for number of errors, since last error might stay same and not be updated.
Might be better to somehow enforce 10min limit and then mark it as failed.

Reply:
Thanks. A timer is better. Propose the name as `item-operation-error-timeout`.

Reply:
@blackpiglet @anshulahuja98 the new parameter `--item-operation-error-timeout` sounds good, will work on a PR for velero controller and then update this PR accordingly.

Reply:
this parameter however might lead to more concerns
where each plugin might want to have a different time limit, whereas this is a global setting

Reply:
It is better if somehow we give this error exit control to the plugin to be smart enough to exit by it's own logic than a global setting

Reply:
Jus realized I didn't think it through.
It's not possible to have a global timer works in all of the plugins.
`item-operation-error-timeout` works globally, because it's trigger in the Velero server.

If we don't modify the BIA and RIA interface, we cannot pass in the timer to the Progress method.

`item-operation-error-timeout` can only works in the backup_operations_controllers.go. Looks like that is not what we want to do.

Add a default timer in the plugin code is the simplest way, but we cannot modify the timer.
Another possible way is adding the Init method just as the ObjectStore and the VolumeSnapshotter plugins.

Reply:
@shubham-pampattiwar 
That should work.